api/types.go

```diff
@@ -12,21 +15,31 @@ type ProxyConfig struct {
 }
 
 type OIDCConfig struct {
-	IssuerUrl    *string
-	ClientId     *string
-	ClientSecret *string
-	Audience     *string
+	IssuerUrl    *string `json:"issuer_url"`
+	ClientId     *string `json:"client_id"`
+	ClientSecret *string `json:"client_secret"`
+	Audience     *string `json:"audience"`
 	Enabled      *bool
+	Filename     *string
 }
 
 type TokenVerificationConfig struct {
 	Enabled *bool
 	Url     *string
 }
 
-func (c *OIDCConfig) Validate() {
+func (c *OIDCConfig) ReadAndValidate() {
 	if !*c.Enabled {
 		return
+	} else {
+		configFile, err := os.Open(*c.Filename)
+		if err != nil {
+			panic(err)
+		}
+		data, err := io.ReadAll(configFile)
+		if err == nil && data != nil {
+			err = json.Unmarshal(data, &*c)
```

Comment:
Isn't `&*c` the same as `c` ?

Reply:
@valeriiashapoval your first dereferencing the pointer and then taking the address of the dereferenced value - this is equivalent to just passing the pointer in the first place